packages/scaffold-config/src/frameworks.ts

```diff
@@ -61,11 +61,9 @@ function getBundlerDependency (bundler: WizardBundler['type'], projectPath: stri
   }
 }
 
-export const WIZARD_MOUNT_MODULES = ['cypress/react', 'cypress/react18', 'cypress/vue', 'cypress/vue2', 'cypress/angular'] as const
+export type WizardMountModule = Awaited<ReturnType<typeof WIZARD_FRAMEWORKS[number]['mountModule']>>
```

Comment:
Example using the new `Awaited` utility type.

Reply:
🎉

Reply:
For fun, if you do "go to definition" on `Awaited` you can see how they built it:

```ts
type Awaited<T> =
    T extends null | undefined ? T : // special case for `null | undefined` when not in `--strictNullChecks` mode
        T extends object & { then(onfulfilled: infer F): any } ? // `await` only unwraps object types with a callable `then`. Non-object types are not unwrapped
            F extends ((value: infer V, ...args: any) => any) ? // if the argument to `then` is callable, extracts the first argument
                Awaited<V> : // recursively unwrap the value
                never : // the argument to `then` was not callable
        T; // non-object or non-thenable
```

Reply:
🤯 would take me a whole day to try and cook up something like that from scratch

Reply:
It's quite cool, notice they use `Awaited` in the definition? Recursive types!

This is actually a great type to study... you can learn

- extends
- conditionals
- infer
- recursive types
- never (equivalent of `throw Error('this is impossible')` for types